### ISSUE TITLE
Compiler warning cased by boost_multi

### ISSUE THREAD

**Issue by ye-luo** [QMCPACK/qmcpack#1272]
We got a lot of compiler warnings from boost_multi.
see cdash nightly section. Intel compiler + boost 1.61. I feel that clang does complain anything probably because the corresponding warning may have been turned off.
https://cdash.qmcpack.org/CDash/viewBuildError.php?type=1&buildid=35088
`
/.../qmcpack-develop/external_codes/boost_multi/multi/./array_ref.hpp(893): warning #488: template parameter "Extensions" is not used in declaring the parameter types of function template "boost::multi::const_array_ref::const_array_ref(ElementPtr, const boost::multi::layout_t &)"
`

**Comment by correaa** [comment]
This is fixed in an upcomming version of [Boost.]Multi, we can wait for that or at the moment simply remove the `..., typename Extensions>` from the set of template parametes. 

replace:
```
	template<class Extensions>
	constexpr const_array_ref(ElementPtr p, layout_t<D> const& lo)
	: basic_array<T, D, ElementPtr>{p, lo}
	{}
```
with:
```
	constexpr const_array_ref(ElementPtr p, layout_t<D> const& lo)
	: basic_array<T, D, ElementPtr>{p, lo}
	{}
```

**Comment by prckent** [comment]
@correaa Do you plan to update the code to avoid the warning? So many are generated I am in favor of doing so now. However perhaps there are extra subtleties?

**Comment by ye-luo** [comment]
This issue is open against qmcpack. Unless compiling qmcpack stops reporting warning, keep this issue open at the moment.

**Comment by ye-luo** [comment]
Resolved by #1360
